soem/ethercatconfig.c

```diff
@@ -482,8 +483,8 @@ int ecx_config_init(ecx_contextt *context, uint8 usetable)
             context->slavelist[slave].SM[1].StartAddr = htoes(context->slavelist[slave].mbx_ro);
             context->slavelist[slave].SM[1].SMlength = htoes(context->slavelist[slave].mbx_rl);
             context->slavelist[slave].SM[1].SMflags = htoel(EC_DEFAULTMBXSM1);
-            context->slavelist[slave].mbx_proto =
-               ecx_readeeprom2(context, slave, EC_TIMEOUTEEP);
+            val = ecx_readeeprom2(context, slave, EC_TIMEOUTEEP);
```

Comment:
skip adding uint32_t val, use the same look'n'feel as done earlier in the file, for example L388 "eedat = etohl(ecx_readeeprom2(context, slave, EC_TIMEOUTEEP)); /* read mailbox offset */"

Reply:
I used the temporary variable because `etohs()` is a macro. Fully expanded, it results in (on big-endian systems):
`((((uint32)(ecx_readeeprom2(context, slave, 20000)) & 0xff000000) >> 24) | \ (((uint32)(ecx_readeeprom2(context, slave, 20000)) & 0x00ff0000) >> 8) | \ (((uint32)(ecx_readeeprom2(context, slave, 20000)) & 0x0000ff00) << 8) | \ (((uint32)(ecx_readeeprom2(context, slave, 20000)) & 0x000000ff) << 24))`.
So `ecx_readeeprom2()` would called 4 times, instead of once.

Reply:
Ok, that's no good, that would need to be addressed.  A few of those and other ethercat commands but isolated to ethercatconfig.c, would you mind fixing it in this PR?

Reply:
Done in new commit